### PR TITLE
Fix: In-app changelog prompt does not appear if category-based filtering results in an empty version

### LINKED ISSUE #2619: In-app changelog prompt should appear if category-based filtering has left it empty
We have a feature in `ChangelogService` that supports a config key to exclude certain categories from the in-app display of the log.  (So we can add e.g. Technical changes in the log but not have business users see and be confused by them.)

https://github.com/xh/hoist-react/blob/7cf837e2adbe5bc763328d2b863c36173bc16cc3/svc/ChangelogService.js#L148

However, it can be the case that use of this feature will cause _all_ entries for a given version to be filtered out, resulting in the user being prompted with the 🎁  icon to open the log and seeing a broken-looking display with no bullet points.

The service should take the filter into account when deciding if a version effectively "exists" before showing the prompt or printing the version header / date in the log.

### PR DESCRIPTION
**Pull request by saba-mo**
Closes #2619 

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by lbwexler** [comment]
Saba and I went over this, including how to test.  This solution seems fine, and works as requested.

Its a little unfortunate that if you do bring it up now, and the latest changes did *not* include any of your categories, you see an old version number.


We should probably *preserve* the empty versions and:

 1) Show some default text like (Misc. Changes).
 2) Get the lastest non-empty version, and show the gift box icon, only if *that* is out of date. (see line 159)

File: desktop/appcontainer/ChangelogDialog.js [hunk 1]
```diff
@@ -58,19 +58,28 @@ const version = hoistCmp.factory(
...
+                            items: [
+                                h3(cat.title),
+                                ul(...cat.items.map(item => li(item)))
+                            ]
+                        });
+                    })
+
+                ] :
+                [
+                    h2(`v${version.version}`),
```
Comment (lbwexler) [inline desktop/appcontainer/ChangelogDialog.js:77]: Think the title ought to be the same here (line 77 and 63, no?)

**Comment by lbwexler** [comment]
Saba -- please review my latest commit, with some refactoring and slight change in logic regarding when the icon is shown.

**Comment by lbwexler** [comment]
cr: Saba

**Comment by amcclain** [comment]
Just checking - will the  🎁 icon now show if there is a non-empty CL entry for a version which you have yet to look at, even if the version which triggers the prompt is no longer the current app version?

I don't have a strong opinion on it, just confirming if that change was made in case anyone notices / asks about that change. I can see it being reasonable for the feature to behave either which way, and I can see an advantage to still encouraging the user to e.g. read the latest minor (feature) release notes even if a newer patch release was required that did not have a note.

**Comment by saba-mo** [comment]
Yes, the 🎁 icon will show if there is a non-empty version that you have yet to look at, even if it's no longer the current app version. It will only not show if you have already seen all non-empty versions.

**Comment by amcclain** [comment]
Great thanks - makes sense to me! 

🎁 's all around 😄

**Pull request merged**